cirq-core/cirq/transformers/optimize_for_target_gateset.py

```diff
@@ -102,20 +102,29 @@ def optimize_for_target_gateset(
     context: Optional['cirq.TransformerContext'] = None,
     gateset: Optional['cirq.CompilationTargetGateset'] = None,
     ignore_failures: bool = True,
+    max_num_passes: Union[int, None] = 1,
 ) -> 'cirq.Circuit':
     """Transforms the given circuit into an equivalent circuit using gates accepted by `gateset`.
 
+    Repeat max_num_passes times or when `max_num_passes=None` until no further changes can be done
     1. Run all `gateset.preprocess_transformers`
     2. Convert operations using built-in cirq decompose + `gateset.decompose_to_target_gateset`.
     3. Run all `gateset.postprocess_transformers`
 
+    Note:
+        The optimizer is a heuristic and may not produce optimal results even with
+        max_num_passes=None. The prerprocessors and postprocessors of the gate set
+        as well as their order yield different results.
+
+
     Args:
         circuit: Input circuit to transform. It will not be modified.
         context: `cirq.TransformerContext` storing common configurable options for transformers.
         gateset: Target gateset, which should be an instance of `cirq.CompilationTargetGateset`.
         ignore_failures: If set, operations that fail to convert are left unchanged. If not set,
             conversion failures raise a ValueError.
-
+        max_num_passes: The maximum number of passes to do. A value of `None` means to keep
+            iterating until no further improvements can be made.
```

Comment:
"improvement" can be subjective. We should be explicit about the stopping criteria at the very least(reduce number of moments or operations) and maybe eventually provide users with a configurable way to specify the stopping criteria (probably just as another argument)

Reply:
updated to clarify the stopping criteria